docs/dev/how_to/github-runner-setup-guide.md

```diff
@@ -0,0 +1,254 @@
+# Step-by-Step Guide to Create a Self-Hosted GitHub Runner on Azure
+
+This guide provides instructions for setting up self-hosted GitHub runners on both Windows (using PowerShell) and Linux (using Bash and Azure CLI). Self-hosted runners allow you to run GitHub Actions workflows on your own infrastructure.
+
+## Prerequisites
+
+- You need to have Owner rights in the Chef org
+- You need a resource group in the "Engineering Dev/Test" subscription
+- For Windows: Administrative privileges on the Windows machine
+- For Linux: An Azure subscription and Azure CLI installed
+- GitHub Personal Access Token (PAT) with `repo` scope (for repository-level runners) or `admin:org` scope (for organization-level runners)
+
+Github also allows you to create a token to execute the runner with that does not require your PAT. Do this to get a spiffy token:
+
+```PowerShell
+gh api --method POST /repos/chef/chef/actions/runners/registration-token \
+  --header 'Accept: application/vnd.github+json' \
+  --header 'X-GitHub-Api-Version: 2022-11-28'
+```
+
+## Part 1: Windows Runner Setup (Using PowerShell)
```

Comment:
We are using the default Windows runners provided by GHA and not self-hosted runner for testing currently. The only reason we need a Linux selfhosted machine, is FIPS functionality is on OS level needs additional charges e.g for Ubuntu it needs Pro subscription. So may be we can add a note for windows details here that is not being used currently in our CI setup and we use default windows runners provided by Github and this might be a good to know information section

Reply:
Good Point, I added verbiage on lines 5, and 6 to that effect.